src/Build/Evaluation/IProperty.cs

```diff
@@ -35,5 +35,11 @@ string EvaluatedValueEscaped
         {
             get;
         }
+
+        bool IsEnvironmentProperty
+        {
+            get;
+            set;
```

Comment:
do we need a setter here?

do we need this at all? Should we just add this to ProjectPropertyNotXmlBacked?

Reply:
I see we also rely on this in ProjectPropertyInstance too

Reply:
I can add IsEnvironmentProperty to ProjectPropertyInstance and ProjectProperty instead of IProperty, but this seems safer to me should we add a new kind of property. It forces us to consider whether it can be environment-derived and log an event if so.

It was (in my opinion) a really clean way of doing this in a previous iteration. Now, it's more of a push as far as I'm concerned.

I use the setter once in ProjectPropertyInstance as you saw. It means I don't need an extra constructor, but that isn't a great reason. I can take it out if you don't like it.